adapters/repos/db/vector/hnsw/compress.go

```diff
@@ -86,7 +86,45 @@ func (h *hnsw) compress(cfg ent.UserConfig) error {
 		if err != nil {
 			return fmt.Errorf("Compressing vectors: %w", err)
 		}
-		h.commitLog.AddPQ(h.compressor.ExposeFields())
+		h.commitLog.AddCompression(h.compressor.ExposeFields())
+	} else if cfg.SQ.Enabled {
+		if h.isEmpty() {
+			return errors.New("Compress command cannot be executed before inserting some data. Please, insert your data first.")
+		}
+
+		cleanData := make([][]float32, 0, len(data))
+		for i := range data {
```

Comment:
Are we going to have a `trainingLimit` for SQ as well?

Reply:
Yes, already add the check for it. Thanks for pointing this out.